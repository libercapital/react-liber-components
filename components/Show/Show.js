"use strict";

require("core-js/modules/web.dom.iterable");

require("core-js/modules/es6.array.iterator");

require("core-js/modules/es6.object.to-string");

require("core-js/modules/es6.string.iterator");

require("core-js/modules/es6.weak-map");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

require("core-js/modules/es7.symbol.async-iterator");

require("core-js/modules/es6.symbol");

require("core-js/modules/es6.object.define-property");

require("core-js/modules/es6.object.create");

require("core-js/modules/es6.object.set-prototype-of");

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _reactTransitionGroup = require("react-transition-group");

var _SimpleAnime = _interopRequireDefault(require("../SimpleAnime"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function _getRequireWildcardCache() { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; if (obj != null) { var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _typeof(obj) { if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

function _possibleConstructorReturn(self, call) { if (call && (_typeof(call) === "object" || typeof call === "function")) { return call; } return _assertThisInitialized(self); }

function _assertThisInitialized(self) { if (self === void 0) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return self; }

function _getPrototypeOf(o) { _getPrototypeOf = Object.setPrototypeOf ? Object.getPrototypeOf : function _getPrototypeOf(o) { return o.__proto__ || Object.getPrototypeOf(o); }; return _getPrototypeOf(o); }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function"); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, writable: true, configurable: true } }); if (superClass) _setPrototypeOf(subClass, superClass); }

function _setPrototypeOf(o, p) { _setPrototypeOf = Object.setPrototypeOf || function _setPrototypeOf(o, p) { o.__proto__ = p; return o; }; return _setPrototypeOf(o, p); }

var Show =
/*#__PURE__*/
function (_Component) {
  _inherits(Show, _Component);

  function Show() {
    var _this;

    _classCallCheck(this, Show);

    _this = _possibleConstructorReturn(this, _getPrototypeOf(Show).call(this));
    _this.state = {
      transition_ended: false
    };
    return _this;
  }

  _createClass(Show, [{
    key: "render",
    value: function render() {
      var _this2 = this;

      var _this$props = this.props,
          children = _this$props.children,
          enter = _this$props.enter,
          exit = _this$props.exit,
          timeout = _this$props.timeout;

      if ((!this.state.transition_ended || this.props["in"]) && children !== null && children !== undefined) {
        return _react["default"].createElement(_reactTransitionGroup.Transition, {
          onExited: function onExited() {
            setTimeout(function () {
              _this2.setState({
                transition_ended: true
              });
            }, timeout + 50);
          },
          mountOnEnter: true,
          timeout: timeout,
          "in": this.props["in"]
        }, function (state) {
          if (state === 'entering') return null;

          if (state === 'entered' && enter) {
            return _react["default"].createElement(_SimpleAnime["default"], {
              opacity: [0, 1],
              duration: timeout,
              translateX: ['-50px', '0px'],
              easing: "easeOutElastic"
            }, _react["default"].createElement("div", null, children));
          }

          if (state === 'exited' && exit) {
            return _react["default"].createElement(_SimpleAnime["default"], {
              opacity: [1, 0],
              duration: timeout,
              translateX: ['0px', '-50px'],
              easing: "easeInElastic"
            }, _react["default"].createElement("div", null, children));
          }

          return children;
        });
      }

      return null;
    }
  }], [{
    key: "getDerivedStateFromProps",
    value: function getDerivedStateFromProps() {
      return {
        transition_ended: false
      };
    }
  }]);

  return Show;
}(_react.Component);

Show.propTypes = {
  children: _propTypes["default"].element.isRequired,
  "in": _propTypes["default"].bool.isRequired,
  enter: _propTypes["default"].bool,
  exit: _propTypes["default"].bool,
  timeout: _propTypes["default"].number
};
Show.defaultProps = {
  enter: true,
  exit: true,
  timeout: 500
};
var _default = Show;
exports["default"] = _default;