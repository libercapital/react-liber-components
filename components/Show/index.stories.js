"use strict";

require("core-js/modules/es6.object.define-property");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.defaultStory = exports["default"] = void 0;

require("core-js/modules/es6.regexp.to-string");

require("core-js/modules/es6.date.to-string");

require("core-js/modules/es6.object.to-string");

require("core-js/modules/es7.symbol.async-iterator");

require("core-js/modules/es6.symbol");

require("core-js/modules/web.dom.iterable");

require("core-js/modules/es6.array.is-array");

var _react = _interopRequireDefault(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _Show = _interopRequireDefault(require("./Show"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _nonIterableRest(); }

function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance"); }

function _iterableToArrayLimit(arr, i) { if (!(Symbol.iterator in Object(arr) || Object.prototype.toString.call(arr) === "[object Arguments]")) { return; } var _arr = []; var _n = true; var _d = false; var _e = undefined; try { for (var _i = arr[Symbol.iterator](), _s; !(_n = (_s = _i.next()).done); _n = true) { _arr.push(_s.value); if (i && _arr.length === i) break; } } catch (err) { _d = true; _e = err; } finally { try { if (!_n && _i["return"] != null) _i["return"](); } finally { if (_d) throw _e; } } return _arr; }

function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }

var ContentCenter = _styledComponents["default"].div.withConfig({
  displayName: "indexstories__ContentCenter",
  componentId: "zseur0-0"
})(["width:100%;height:100%;display:flex;justify-content:center;align-items:center;padding:50px;box-sizing:border-box;"]);

var _default = {
  title: 'Components 1.0|Show',
  parameters: {
    jest: ['Show']
  }
};
exports["default"] = _default;

var defaultStory = function defaultStory() {
  var _React$useState = _react["default"].useState(false),
      _React$useState2 = _slicedToArray(_React$useState, 2),
      state = _React$useState2[0],
      setState = _React$useState2[1];

  return _react["default"].createElement(_react["default"].Fragment, null, _react["default"].createElement("button", {
    type: "button",
    onClick: function onClick() {
      return setState(!state);
    }
  }, "action"), _react["default"].createElement(_Show["default"], {
    "in": state,
    timeout: 500
  }, _react["default"].createElement(ContentCenter, null, "Conteudo")));
};

exports.defaultStory = defaultStory;
defaultStory.story = {
  name: 'default'
};