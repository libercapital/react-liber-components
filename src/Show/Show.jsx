import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Transition } from 'react-transition-group';
import Anime from '../SimpleAnime';

class Show extends Component {
  constructor() {
    super();
    this.state = {
      transition_ended: false,
    };
  }

  static getDerivedStateFromProps() {
    return { transition_ended: false };
  }

  render() {
    const { children, enter, exit, timeout } = this.props;
    if (
      (!this.state.transition_ended || this.props.in) &&
      (children !== null && children !== undefined)
    ) {
      return (
        <Transition
          onExited={() => {
            setTimeout(() => {
              this.setState({ transition_ended: true });
            }, timeout + 50);
          }}
          mountOnEnter
          timeout={timeout}
          in={this.props.in}
        >
          {state => {
            if (state === 'entering') return null;
            if (state === 'entered' && enter) {
              return (
                <Anime
                  opacity={[0, 1]}
                  duration={timeout}
                  translateX={['-50px', '0px']}
                  easing="easeOutElastic"
                >
                  <div>{children}</div>
                </Anime>
              );
            }
            if (state === 'exited' && exit) {
              return (
                <Anime
                  opacity={[1, 0]}
                  duration={timeout}
                  translateX={['0px', '-50px']}
                  easing="easeInElastic"
                >
                  <div>{children}</div>
                </Anime>
              );
            }
            return children;
          }}
        </Transition>
      );
    }
    return null;
  }
}

Show.propTypes = {
  children: PropTypes.element.isRequired,
  in: PropTypes.bool.isRequired,
  enter: PropTypes.bool,
  exit: PropTypes.bool,
  timeout: PropTypes.number,
};

Show.defaultProps = {
  enter: true,
  exit: true,
  timeout: 500,
};

export default Show;
