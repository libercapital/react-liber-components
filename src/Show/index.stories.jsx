import React from 'react';
import styled from 'styled-components';
import Show from './Show';

const ContentCenter = styled.div`
  width: 100%;
  height: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 50px;
  box-sizing: border-box;
`;

export default {
  title: 'Components 1.0|Show',

  parameters: {
    jest: ['Show'],
  },
};

export const defaultStory = () => {
  const [state, setState] = React.useState(false);
  return (
    <>
      <button type="button" onClick={() => setState(!state)}>
        action
      </button>
      <Show in={state} timeout={500}>
        <ContentCenter>Conteudo</ContentCenter>
      </Show>
    </>
  );
};

defaultStory.story = {
  name: 'default',
};
