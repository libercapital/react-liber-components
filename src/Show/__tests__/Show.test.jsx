import React from 'react';
import { mount, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import 'jest-styled-components';
import Show from '../Show';

jest.useFakeTimers();

describe('Testing SelectField Component', () => {
  let props;
  configure({ adapter: new Adapter() });

  beforeEach(() => {
    props = {
      in: false,
      children: (<div />),
    };
  });
  it('should render correctly closed', () => {
    const wrapper = mount(<Show {...props} />);
    jest.runAllTimers();
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should render correctly open', () => {
    props.in = true;
    const wrapper = mount(<Show {...props} />);
    jest.runAllTimers();
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should render correctly when opened', () => {
    const wrapper = mount(<Show {...props} />);
    wrapper.setProps({ in: true });
    jest.runAllTimers();
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should render correctly when closed', () => {
    props.in = true;
    const wrapper = mount(<Show {...props} />);
    wrapper.setProps({ in: false });
    jest.runAllTimers();
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
