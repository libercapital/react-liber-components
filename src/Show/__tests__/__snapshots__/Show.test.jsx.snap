// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing SelectField Component should render correctly closed 1`] = `
<Show
  enter={true}
  exit={true}
  in={false}
  timeout={500}
>
  <Transition
    appear={false}
    enter={true}
    exit={true}
    in={false}
    mountOnEnter={true}
    onEnter={[Function]}
    onEntered={[Function]}
    onEntering={[Function]}
    onExit={[Function]}
    onExited={[Function]}
    onExiting={[Function]}
    timeout={500}
    unmountOnExit={false}
  />
</Show>
`;

exports[`Testing SelectField Component should render correctly open 1`] = `
<Show
  enter={true}
  exit={true}
  in={true}
  timeout={500}
>
  <Transition
    appear={false}
    enter={true}
    exit={true}
    in={true}
    mountOnEnter={true}
    onEnter={[Function]}
    onEntered={[Function]}
    onEntering={[Function]}
    onExit={[Function]}
    onExited={[Function]}
    onExiting={[Function]}
    timeout={500}
    unmountOnExit={false}
  >
    <SimpleAnime
      duration={500}
      easing="easeOutElastic"
      opacity={
        Array [
          0,
          1,
        ]
      }
      translateX={
        Array [
          "-50px",
          "0px",
        ]
      }
    >
      <div>
        <div />
      </div>
    </SimpleAnime>
  </Transition>
</Show>
`;

exports[`Testing SelectField Component should render correctly when closed 1`] = `
<Show
  enter={true}
  exit={true}
  in={false}
  timeout={500}
>
  <Transition
    appear={false}
    enter={true}
    exit={true}
    in={false}
    mountOnEnter={true}
    onEnter={[Function]}
    onEntered={[Function]}
    onEntering={[Function]}
    onExit={[Function]}
    onExited={[Function]}
    onExiting={[Function]}
    timeout={500}
    unmountOnExit={false}
  >
    <SimpleAnime
      duration={500}
      easing="easeOutElastic"
      opacity={
        Array [
          0,
          1,
        ]
      }
      translateX={
        Array [
          "-50px",
          "0px",
        ]
      }
    >
      <div>
        <div />
      </div>
    </SimpleAnime>
  </Transition>
</Show>
`;

exports[`Testing SelectField Component should render correctly when opened 1`] = `
<Show
  enter={true}
  exit={true}
  in={true}
  timeout={500}
>
  <Transition
    appear={false}
    enter={true}
    exit={true}
    in={true}
    mountOnEnter={true}
    onEnter={[Function]}
    onEntered={[Function]}
    onEntering={[Function]}
    onExit={[Function]}
    onExited={[Function]}
    onExiting={[Function]}
    timeout={500}
    unmountOnExit={false}
  >
    <SimpleAnime
      duration={500}
      easing="easeInElastic"
      opacity={
        Array [
          1,
          0,
        ]
      }
      translateX={
        Array [
          "0px",
          "-50px",
        ]
      }
    >
      <div>
        <div />
      </div>
    </SimpleAnime>
  </Transition>
</Show>
`;
